presto-router/src/test/java/com/facebook/presto/router/TestScheduler.java

```diff
@@ -69,33 +76,65 @@ public void testUserHashScheduler()
         Scheduler scheduler = new UserHashScheduler();
         scheduler.setCandidates(servers);
 
-        URI target1 = scheduler.getDestination("test").orElse(new URI("invalid"));
-        assertTrue(servers.contains(target1));
+        String[] users = {"test", "user", "1234"};
+        for (String user : users) {
+            URI target1 = scheduler.getDestination(user).orElse(new URI("invalid"));
+            assertTrue(servers.contains(target1));
 
-        URI target2 = scheduler.getDestination("test").orElse(new URI("invalid"));
-        assertTrue(servers.contains(target2));
+            URI target2 = scheduler.getDestination(user).orElse(new URI("invalid"));
+            assertTrue(servers.contains(target2));
 
-        assertEquals(target2, target1);
+            URI target3 = scheduler.getDestination(user).orElse(new URI("invalid"));
+            assertTrue(servers.contains(target3));
+
+            assertEquals(target2, target1);
+            assertEquals(target3, target2);
+        }
```

Comment:
If these assertions are all the same for each user, let's use a DataProvider for this test instead